MyChart.js

```diff
@@ -0,0 +1,154 @@
+
+        const postData = `id=${siteId}&vt=${vt}&dept=${dept}&view=plain&start=${startDateFormatted}&filters=%7B%22Providers%22%3A%7B%2213300632%22%3Atrue%7D%2C%22Departments%22%3A%7B%22133001025%22%3Atrue%7D%2C%22DaysOfWeek%22%3A%7B%220%22%3Atrue%2C%221%22%3Atrue%2C%222%22%3Atrue%2C%223%22%3Atrue%2C%224%22%3Atrue%2C%225%22%3Atrue%2C%226%22%3Atrue%7D%2C%22TimesOfDay%22%3A%22both%22%7D`;
+        const options = {
+            hostname: 'mychartos.ochin.org',
+            port: 443,
+            path:
+                '/mychart/OpenScheduling/OpenScheduling/GetOpeningsForProvider',
+            method: 'POST',
+            headers: {
+                'Content-Type':
+                    'application/x-www-form-urlencoded; charset=UTF-8',
+                'Content-Length': postData.length,
+                __RequestVerificationToken: verificationToken,
+                Cookie: cookie,
+            },
+        };
+
+        // now request the schedule
+        const schedulePromise = new Promise(resolve => {
+            const req = https.request(options, res => {
+                let body = '';
+                responseHeaders = res.headers;
+                res.on('data', chunk => {
+                    body += chunk;
+                });
+                res.on('end', () => {
+                    if (res.statusCode >= 200 && res.statusCode < 300) {
+                        resolve(JSON.parse(body));
+                    } else {
+                        console.error(
+                            'Error status code returned from ' + res.url
+                        );
+                        resolve();
+                    }
+                });
+            });
+            req.write(postData);
+            req.on('error', e => {
+                console.error('Error making scheduling request : ' + e);
+            });
+            req.end();
+        });
+
+        const postResponse = await schedulePromise;
+        if (postResponse) {
+            allDays = postResponse.AllDays;
+
+            // The AllDays property is an object, not an array.
+            // The property of the object is generated from the date, which we don't know
+            // We need to iterate over all the properties to determine the dates and the slots.
+            for (const dayEntryProperty in allDays) {
+                if (
+                    Object.prototype.hasOwnProperty.call(
+                        allDays,
+                        dayEntryProperty
+                    )
+                ) {
+                    dayEntry = allDays[dayEntryProperty];
+                    const date = new Date(dayEntry.DateISO);
+                    const slots = dayEntry.Slots;
+                    const slotsAvailable = slots.length || 0;
+
+                    if (date >= new Date()) {
+                        results.availability[
+                            `${date.getMonth() + 1}/${date.getDate() +
+                                1}/${date.getFullYear()}`
```

Comment:
I have a sneaking suspicion that the getDate() + 1 in conjunction with whatever is returned from dayEntry.DateISO (line 129) is what's causing our bug... we shouldn't have to be adding 1 to getDate(), it's not 0-indexed. I don't know why alarm bells didn't start ringing in my head earlier...